Frameworks/AddressBook/ABMultiValueInternal.mm

```diff
@@ -0,0 +1,164 @@
+//******************************************************************************
+//
+// Copyright (c) 2016 Microsoft Corporation. All rights reserved.
+//
+// This code is licensed under the MIT License (MIT).
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//******************************************************************************
+
+#import "ABMultiValueInternal.h"
+
+// Private helper class to represent the values
+// in an ABMultiValue, which are pairs of a label
+// and value (along with an identifier).
+@interface __ABMultiValuePair : NSObject
+
+@property NSString* label;
+@property id value;
+@property ABMultiValueIdentifier identifier;
+
+- (id)initWithLabel:(NSString*)label andValue:(id)value withIdentifier:(ABMultiValueIdentifier)identifier;
+
+@end
+
+@implementation __ABMultiValuePair
+
+- (id)initWithLabel:(NSString*)label andValue:(id)value withIdentifier:(ABMultiValueIdentifier)identifier {
+    self = [super init];
+    if (self) {
+        self.label = label;
+        self.value = value;
+        self.identifier = identifier;
+    }
+
+    return self;
+}
+
+@end
+
+@implementation _ABMultiValue {
+    NSMutableArray* _list;
+    int _nextIdentifier;
+    ABPropertyType _propertyType;
+    BOOL _mutable;
+}
+
+- (id)initWithPropertyType:(ABPropertyType)propertyType {
+    self = [super init];
+    if (self) {
+        self->_list = [[NSMutableArray alloc] init];
```

Comment:
I don't think it is required, but I'm using it for consistency/clarity

---
In reply to: [75148511](https://github.com/Microsoft/WinObjC/pull/884#discussion_r75148511) [](ancestors = 75148511)